examples/hmac.rs

```diff
@@ -0,0 +1,64 @@
+#![no_std]
+
+use core::fmt::Write;
+use libtock::debug::print_as_hex;
+use libtock::hmac::{HmacDataBuffer, HmacDestBuffer, HmacKeyBuffer};
+use libtock::result::TockResult;
+use libtock::syscalls;
+
+#[libtock::main]
+async fn main() -> TockResult<()> {
+    let mut drivers = libtock::retrieve_drivers()?;
+
+    let mut console = drivers.console.create_console();
+
+    writeln!(console, "Starting HMAC example")?;
+
+    let hmac_driver = drivers.hmac.init_driver()?;
+
+    writeln!(console, "Loading in 0 key")?;
+    let mut key_buffer = HmacKeyBuffer::default();
+    let _key_buffer = hmac_driver.init_key_buffer(&mut key_buffer)?;
+    writeln!(console, "  done")?;
+
+    writeln!(console, "Creating data buffer")?;
+    let mut data_buffer = HmacDataBuffer::default();
+    let data = [
+        'A', ' ', 'l', 'a', 'n', 'g', 'u', 'a', 'g', 'e', ' ', 'e', 'm', 'p', 'o', 'w', 'e', 'r',
+        'i', 'n', 'g', ' ', 'e', 'v', 'e', 'r', 'y', 'o', 'n', 'e', ' ', 't', 'o', ' ', 'b', 'u',
+        'i', 'l', 'd', ' ', 'r', 'e', 'l', 'i', 'a', 'b', 'l', 'e', ' ', 'a', 'n', 'd', ' ', 'e',
+        'f', 'f', 'i', 'c', 'i', 'e', 'n', 't', ' ', 's', 'o', 'f', 't', 'w', 'a', 'r', 'e', '.',
+    ];
+
+    for (i, d) in data.iter().enumerate() {
+        data_buffer.buffer[i] = *d as u8;
+    }
+    let _data_buffer = hmac_driver.init_data_buffer(&mut data_buffer)?;
+    writeln!(console, "  done")?;
+
+    writeln!(console, "Creating dest buffer")?;
+    let mut dest_buffer = HmacDestBuffer::default();
+    let dest_buffer = hmac_driver.init_dest_buffer(&mut dest_buffer)?;
+    writeln!(console, "  done")?;
+
+    let mut temp_buffer = [0; libtock::hmac::DEST_BUFFER_SIZE];
+
+    let mut callback = |_result, _digest| {
+        dest_buffer.read_bytes(&mut temp_buffer[..]);
+
+        for buf in temp_buffer.iter().take(libtock::hmac::DEST_BUFFER_SIZE) {
+            print_as_hex(*buf as usize);
```

Comment:
I would prefer not using the debug methods here (which bypass the protection from the drivers) but using the console instead.

Reply:
The problem with using console is that console is then mutably borrowed multiple times and I get
```
cannot borrow `console` as mutable more than once at a time` errors
```

Reply:
That happens because using the console in that way would actually be unsafe. Using a driver from several callbacks or from a callback and the main function is unsafe and using the debug functions is just working around that (they should be unsafe imho).

Reply:
An alternative would be: set up a`Cell` and let the callback copy the information you want to print there. In the main function wait for the info to be copied, deregister the callback and print the info - or use futures.

Reply:
I have swapped this to using the `write!` macro